HomeWork_9/hw1.py

```diff
@@ -19,22 +19,16 @@
 
 
 def merge_sorted_files(file_list: List[Union[Path, str]]) -> Iterator:
-    def read_next_numb_from_file(file_manager_dict: dict) -> dict:
-        with open(file_manager_dict["filename"], "r") as f:
-            f.seek(file_manager_dict["cursor_position"])
-            numb = f.readline()
-            if numb[:-1].isdigit():
-                file_manager_dict["numb"] = int(numb)
-            else:
-                file_manager_dict["numb"] = None
-            file_manager_dict["cursor_position"] = f.tell()
-            return file_manager_dict
 
     opened_files = {}
-    for file in file_list:
-        opened_files[file] = read_next_numb_from_file(
-            {"filename": file, "cursor_position": 0, "numb": None}
-        )
+    for filename in file_list:
+        file = open(filename, "r")
+        opened_files[filename] = {"file": file}
+        numb = next(opened_files[filename]["file"])
+        if numb[:-1].isdigit():
+            opened_files[filename]["numb"] = int(numb[:-1])
+        else:
+            raise ValueError("String is not a numb")
```

Comment:
Since VvalueError raised anyway we don't need checks.
```suggestion
        opened_files[filename]["numb"] = int(numb[:-1])
```